projects/kit/components/input-date-multi/input-date-multi.component.ts

```diff
@@ -0,0 +1,339 @@
+        return this.isMobile && !this.editable;
+    }
+
+    get computedMin(): TuiDay {
+        return this.min ?? this.options.min;
+    }
+
+    get computedMax(): TuiDay {
+        return this.max ?? this.options.max;
+    }
+
+    get nativeFocusableElement(): HTMLInputElement | null {
+        return this.textfield?.nativeFocusableElement || null;
+    }
+
+    get focused(): boolean {
+        return !!this.textfield?.focused;
+    }
+
+    get computedMobile(): boolean {
+        return this.isMobile && !!this.mobileCalendar;
+    }
+
+    get calendarIcon(): TuiInputDateOptions['icon'] {
+        return this.options.icon;
+    }
+
+    get computedActiveYearMonth(): TuiMonth {
+        return (
+            this.month ||
+            this.value[this.value.length - 1] ||
+            tuiDateClamp(this.defaultActiveYearMonth, this.computedMin, this.computedMax)
+        );
+    }
+
+    get nativeValue(): string {
+        return this.nativeFocusableElement?.value ?? '';
+    }
+
+    set nativeValue(value: string) {
+        if (!this.nativeFocusableElement) {
+            return;
+        }
+
+        this.nativeFocusableElement.value = value.toString();
+    }
+
+    get canOpen(): boolean {
+        return this.interactive && !this.computedMobile;
+    }
+
+    onIconClick(): void {
+        if (!this.computedMobile || !this.mobileCalendar) {
+            return;
+        }
+
+        this.dialogs
+            .open<readonly TuiDay[]>(
+                new PolymorpheusComponent(this.mobileCalendar, this.injector),
+                {
+                    size: 'fullscreen',
+                    closeable: false,
+                    data: {
+                        single: false,
+                        min: this.min,
+                        max: this.max,
+                        disabledItemHandler: this.disabledItemHandler,
+                    },
+                },
+            )
+            .pipe(takeUntil(this.destroy$))
+            .subscribe(value => {
+                this.internalValue = value;
+                this.value = value;
+            });
+    }
+
+    onEnter(search: string): void {
+        if (!this.tagValidator(search)) {
+            return;
+        }
+
+        this.internalValue = tuiImmutableUpdateInputDateMulti(
+            this.internalValue,
+            TuiDay.normalizeParse(search),
+        );
+
+        if (this.inputTag) {
+            this.inputTag.search = '';
+        }
+
+        this.done();
+    }
+
+    onValueChange(value: readonly TuiDay[]): void {
+        this.control?.updateValueAndValidity({emitEvent: false});
+
+        if (!value.length) {
+            this.onOpenChange(true);
+        }
+
+        this.internalValue = value;
+        this.value = value;
+    }
+
+    onDayClick(value: TuiDay): void {
+        this.internalValue = tuiImmutableUpdateInputDateMulti(this.internalValue, value);
+    }
+
+    done(): void {
+        this.value = this.internalValue;
```

Comment:
I don't think we need to track `internalValue`. I think `Done` button is just to close the dropdown, value can be updated as the user clicks.

Reply:
fixed